test/langtools/tools/javac/patterns/BindingsTest1.java

```diff
@@ -25,8 +25,8 @@
  * @test
  * @bug 8231827
  * @summary Basic tests for bindings from instanceof
- * @compile --enable-preview -source ${jdk.version} BindingsTest1.java
- * @run main/othervm --enable-preview BindingsTest1
+ * @compile BindingsTest1.java
```

Comment:
the @compile can be removed

Reply:
I guess I prefer to keep @compile for javac non-programatic tests. The reason is that, for these tests, we can expect we modify javac, but not the test. And if @build (implicit or explicit) is used, the test will not be re-compiled unless their timestamp changes. So we can actually execute an older (already built) version of the test's class file, instead of a newly built class file, which would reflect the changes in javac. So, for tests like this, I prefer to have an explicit @compile, as that ensures the file is compiled every time, regardless how the working directories are set-up, etc.